virttest/utils_misc.py

```diff
@@ -1697,7 +1700,7 @@ def read_from_node_meminfo(self, node_id, key):
         :return: The value in KB
         :rtype: string
         """
-        return self.get_all_node_meminfo()[node_id][key]
+        return self.online_nodes_meminfo[node_id][key]
```

Comment:
If it possible that these data are dynamically changed something, during some test, like hot plug, unplug something, thus maybe some users need these functions provide them the real time data?

Reply:
> If it possible that these data are dynamically changed something, during some test, like hot plug, unplug something, thus maybe some users need these functions provide them the real time data?

A good point ~ I think it is possible no matter such a case exist or not.  @huangyum  What's your opinion ?

Reply:
Yes, that's possible as well as online_nodes_withmem and online_nodes_withcpu. In that case, we have to call `self.online_nodes_meminfo = self.get_all_node_meminfo()` first to get latest status.

Reply:
With this, this origin purpose of this patch has gone ~
Let's fix the kernel_interface issue with this patch, thanks!

Reply:
I don't understand. Why do you revert read_from_node_meminfo?

Reply:
> I don't understand. Why do you revert read_from_node_meminfo?

The all_node_meminfo may change after hotplug/unplug,  to get the latest info, we have to call `get_all_node_meminfo` to get the latest online node. Isn't it ?

Reply:
Yes, but it only happens in hotplug/unplug senario. In most cases, it doesn't change.

Reply:
> Yes, but it only happens in hotplug/unplug senario. In most cases, it doesn't change.

The new change is to cover hotplug/unplug, so i revert it.

Reply:
That's not very convincible to me.  The scenario can still be covered without revert.

Reply:
> That's not very convincible to me. The scenario can still be covered without revert.

How can it be covered with hotplug ?  After hotplug/unplug the online nodes may changed, means the `online_nodes_meminfo` may be changed too.

If you are talking about use `get_all_node_meminfo()` in test case, and get special node info from it in test case, that is another usage i think.

Reply:
No, I meat to refresh online_nodes_meminfo by `self.online_nodes_meminfo = self.get_all_node_meminfo()` after hotplug/unplug.
The process is like this:

numaInfo = class NumaInfo()
hotplug/unplug..
numaInfo.online_nodes_meminfo = numaInfo.get_all_node_meminfo()
mem_info_node = numaInfo.read_from_node_meminfo(node, xx)

Reply:
Or we could create a new instance for class NumaInfo after hotplug/unplug.

Reply:
> No, I meat to refresh online_nodes_meminfo by `self.online_nodes_meminfo = self.get_all_node_meminfo()` after hotplug/unplug.
> The process is like this:
> 
> numaInfo = class NumaInfo()
> hotplug/unplug..
> numaInfo.online_nodes_meminfo = numaInfo.get_all_node_meminfo()
> mem_info_node = numaInfo.read_from_node_meminfo(node, xx)

Why refresh `online_nodes_mem_info`, without the refresh we still can get the latest info with `read_from_node_meminfo` ?
For this, my expect usage after hotplug/unplug is:
```
numaInfo = class NumaInfo()
hotplug/unplug
mem_info_node = numaInfo.read_from_node_meminfo(node, xx)
```

Reply:
Then it comes to the original question. You will call get_all_node_meminfo() many times as node number, which is unnecessary.

Reply:
> Then it comes to the original question. You will call get_all_node_meminfo() many times as node number, which is unnecessary.

Eh~  after considering the hotplug/unplug scenario, IMO, this usage already be dropped, that is why i revert the change. and do not plan to use `read_from_node_meminfo`:

Not only one way to avoid the repeated call, and here, i think we need keep consistent, the functions should all use the latest.
In test case, without use `read_from_node_meminfo` is not a great loss:
```
numa_info = NumaInfo(***)
numa_info.online_nodes_meminfo[node_id][key]
```

Reply:
That could work too. I'm not sure which is better. Up to you.

Reply:
> That could work too. I'm not sure which is better. Up to you.

Thanks, let's keep consistent in this class, all use latest, thanks~
Do you have other comments ?

Reply:
Pls see L1714.